pallets/liquidity-pools/src/lib.rs

```diff
@@ -825,6 +842,128 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Block a remote investor from performing investment tasks until lock
+		/// is removed.
+		///
+		/// Origin: Pool admin
+		#[pallet::call_index(14)]
+		#[pallet::weight(T::WeightInfo::freeze_investor())]
+		pub fn freeze_investor(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			tranche_id: T::TrancheId,
+			domain_address: DomainAddress,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+			let local_address = T::DomainAddressToAccountId::convert(domain_address.clone());
+
+			ensure!(
+				T::PoolInspect::pool_exists(pool_id),
+				Error::<T>::PoolNotFound
+			);
+			ensure!(
+				T::PoolInspect::tranche_exists(pool_id, tranche_id),
+				Error::<T>::TrancheNotFound
+			);
+
+			ensure!(
+				T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					who.clone(),
+					Role::PoolRole(PoolRole::PoolAdmin)
+				),
+				Error::<T>::NotPoolAdmin
+			);
+			Self::validate_investor_status(
+				local_address.clone(),
+				pool_id,
+				tranche_id,
+				T::Time::now(),
+				false,
+			)?;
+
+			T::Permission::add(
```

Comment:
Any specific reason this is not implemented in the same way as `update_member`: the permission is set separately using the `permsisions.add_permission` extrinsic, and then in the liquidity pools method it is just checked that the permission is already set?

Reply:
Great point. I will refactor it to be in sync with the `update_member` flow!